src/productTypeInfo.js

```diff
@@ -0,0 +1,75 @@
+const knownProducts = {
+  '358': {
+    model: 'Dyson Pure Humidify+Cool',
+    hardwareRevision: 'PH01',
+    hasAdvancedAirQualitySensors: true,
+    hasHumidifier: true,
+    hasJetFocus: true,
+  },
+  '358E': {
+    model: 'Dyson Pure Humidify+Cool Formaldehyde',
+    hardwareRevision: 'PH04',
+    hasAdvancedAirQualitySensors: true,
+    hasHumidifier: true,
+    hasJetFocus: true,
+  },
+  '438': {
+    model: 'Dyson Pure Cool Tower',
+    hardwareRevision: 'TP04',
+    hasJetFocus: true,
+    hasAdvancedAirQualitySensors: true,
+  },
+  '438E': {
+    model: 'Dyson Pure Cool',
+    hardwareRevision: 'TP07',
+    hasJetFocus: true,
+    hasAdvancedAirQualitySensors: true,
+  },
+  '455': {
+    model: 'Dyson Pure Hot+Cool Link',
+    hardwareRevision: 'HP02',
+    hasHeating: true,
+    hasJetFocus: true,
+  },
+  '469': {
+    model: 'Dyson Pure Cool Link Desk',
+    hardwareRevision: 'DP01',
+  },
+  '475': {
+    model: 'Dyson Pure Cool Link Tower',
+    hardwareRevision: 'TP02',
+  },
+  '520': {
+    model: 'Dyson Pure Cool Desk',
+    hardwareRevision: 'DP04',
+    hasJetFocus: true,
+    hasAdvancedAirQualitySensors: true,
+  },
+  '527': {
+    model: 'Dyson Pure Hot+Cool',
+    hardwareRevision: 'HP04',
+    hasJetFocus: true,
+    hasAdvancedAirQualitySensors: true,
+    hasHeating: true,
+  },
+  '527E': {
+    model: 'Dyson Purifier Hot+Cool Formaldehyde',
+    hardwareRevision: 'HP09',
+    hasJetFocus: true,
+    hasAdvancedAirQualitySensors: true,
+    hasHeating: true,
+  },
+};
+
+const clone = (data) => JSON.parse(JSON.stringify(data));
```

Comment:
I like the idea of separating the product info from the logic code! However, the cloning approach seems a bit complicated. What do you think about using a switch:

```
switch (productType) {
  case '358':
    return {
      model: 'Dyson Pure Humidify+Cool',
      hardwareRevision: 'PH01',
      hasAdvancedAirQualitySensors: true,
      hasHumidifier: true,
      hasJetFocus: true
    };
  ...
}
```

Reply:
Thanks for the feedback! I pushed a new change now, where I simply copy the attributes from a `knownProducts` item into a new object. The reason why I'm not a huge fan of the switch() approach is that it's a bit messy to add default values afterwards. This can be fixed however by adding another private function that returns the know attributes, and then apply the defaults afterwards. Please let me know what you prefer.

Reply:
I like your approach, much better than cloning via JSON serialization!